SlackAPI/SlackSocketClient.cs

```diff
@@ -22,31 +21,37 @@ public class SlackSocketClient : SlackClient
         public bool IsConnected { get { return underlyingSocket != null && underlyingSocket.Connected; } }
 
         public event Action OnHello;
-		internal LoginResponse loginDetails;
+        private LoginResponse loginDetails;
 
         public SlackSocketClient(string token)
             : base(token)
         {
+        }
 
+        public SlackSocketClient(string token, IWebProxy proxySettings)
+            : base(token, proxySettings)
+        {
         }
 
-		public override void Connect(Action<LoginResponse> onConnected, Action onSocketConnected = null)
-		{
-			base.Connect((s) => {
-				ConnectSocket(onSocketConnected);
-				onConnected(s);
-			});
-		}
+        public override void Connect(Action<LoginResponse> onConnected, Action onSocketConnected = null)
+        {
+            base.Connect((s) => {
+                if (s.ok)
+                    ConnectSocket(onSocketConnected);
```

Comment:
Don't try to connect the websocket if we were not able to connect the SlackClient instance (or it will raise a NullReferenceException in a background thread and the callback is never called. It's related to #107)